Lib/idlelib/debugger.py

```diff
@@ -130,80 +154,187 @@ def make_gui(self):
+
+    def enable_buttons(self, buttons=None):
+        for key in self.buttons:
+            if buttons is None or not key in buttons:
+                self.buttondata[key][0]['image'] = self.buttondata[key][2]
+                self.buttondata[key][0]['foreground'] = '#aaaaaa'
+                self.buttondata[key][0]['cursor'] = ''
+                self.buttondata[key][0].bind('<1>', 'break')
+                self.buttondata[key][0].bind('<<context-menu>>', 'break')
+            else:
+                self.buttondata[key][0]['image'] = self.buttondata[key][1]
+                self.buttondata[key][0]['foreground'] = '#000000'
+                self.buttondata[key][0].bind('<1>', self.button_cmds[key])
+                self.buttondata[key][0].bind('<<context-menu>>',
+                                             self.button_cmds[key])
+                self.buttondata[key][0]['cursor'] = self.clickable_cursor
+
+    def stack_selection_changed(self):
+        self.show_vars()
+
+    def stack_doubleclick(self):
+        sel = self.stack.selection()
+        if len(sel) == 1:
+            self.show_source(sel[0])
+
+    def stack_contextmenu(self, event):
+        item = self.stack.identify('item', event.x, event.y)
+        if item is not None and item != -1 and item != '':
+            menu = Menu(self.top, tearoff=0)
+            menu.add_command(label='View Source',
+                             command = lambda: self.show_source(item))
+            menu.tk_popup(event.x_root, event.y_root)
+
+    def show_source(self, item):
+        if item in self.framevars:
+            fname = self.framevars[item][2]
+            lineno = self.framevars[item][3]
+            if fname[:1] + fname[-1:] != "<>" and os.path.exists(fname):
+                self.flist.gotofileline(fname, lineno)
+
+    def show_status(self, msg, error=False):
+        self.status['text'] = msg
+        self.status['foreground'] = '#ff0000' if error else '#006600'
+        self.status['font'] = self.status_error_font if error \
+                            else self.status_normal_font
+
+    def clear_stack(self):
+        self.stack.delete(*self.stack.get_children(''))
+        self.vars.delete(*self.vars.get_children(self.locals))
+        self.vars.delete(*self.vars.get_children(self.globals))
+        self.vars.detach(self.locals)
+        self.vars.detach(self.globals)
+        self.var_values = {}
+
+    def add_stackframe(self, frame, lineno, current=False):
+        func = frame.f_code.co_name
+        if func in ("?", "", None):
+            func = '.'
+        try:
+            selfval = frame.f_locals['self']
+            if selfval.__class__.__name__ == 'str':
+                # we've probably got the string representation of the
+                # object sent from the remote debugger, see if we can
+                # parse it into something useful
+                match = re.match('^<(?:.*)\.([^\.]*) object at 0x[0-9a-f]+>$',
```

Comment:
Special chars are normal inside [] without \, so I think [^\.] and [^.] are the same (I prefer the latter) so that [^\\.] is needed to match 3 all 3 chars. Prefix r suppress strings escapes, but there are none, so I think it has no effect, though some prefer to use it for all REs.  Outside the brackets, '\.' matches literal period, which is what want, whereas '\\.' matches multiple backslashes?  I am not sure why later periods are excluded.   CPython now prints addresses with A-F.

In any case, non-trivial REs should be defined at the top of the file so that they can be tested.

Reply:
Sorry Terry, I don't quite understand what you mean in a couple of places:

> I think [^.] and [^.] are the same

What did you mean to say because those read the same?!

> Outside the brackets, '.' matches literal period

Am I misunderstanding where you are referring to because I believe a `.` "matches any character except a newline" except when in the square brackets (in which case it is literal).

Reply:
What you read is not what I wrote because unknown to me, markup interprets or removes `\`s.  I edited and added backticks \`  \` to prevent that, so read again.

Reply:
Resolved